src/helpers/getExternalQuestStatus.ts

```diff
@@ -0,0 +1,34 @@
+import { QuestLogInfo } from '@hyperplay/ui'
+import { ExternalEligibility, Quest } from '@hyperplay/utils'
+import { canClaimLeaderboardReward } from './canClaimReward'
+import dayjs from 'dayjs'
+
+export function getExternalQuestStatus(
+  quest: Quest,
+  externalEligibility: ExternalEligibility | null
+): QuestLogInfo['state'] | undefined {
+  const questEnded = quest.end_date
+    ? dayjs(quest.end_date).isBefore(dayjs())
+    : false
+
+  if (quest.status === 'COMPLETED') {
+    return undefined
+  }
```

Comment:
I feel like we need another section on the quest log for these types of quests that are just waiting on eligibility data finalization, but that's probably too much scope creep.

Also I think we should return `undefined` to hide the quest only for `'INACTIVE'` quests or `CLAIMABLE` quests that the user isn't eligible for (like you have below) right? `COMPLETED` quests should still be visible to the user in the quest log in the client to support this design (just the quest details part) https://www.figma.com/design/yHrBPoE70k1I2icW4paEit/Quests?node-id=1382-53700&t=IQCfck3Z0S0pJKoj-4

Reply:
Let me double-check that with the product team.

Reply:
Actually, my current understanding of that quest is that its state is `ACTIVE`, but the quest has ended.

Reply:
This was an incorrect assumption, and you're right to assume the finalizing state is the same as completed. I will update this.

Reply:
Updated the code to reflect he changes.